x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -862,6 +865,10 @@ boolean hasQuery() {
             return query != null;
         }
 
+        public boolean checkSelector(IndexComponentSelector selector) {
```

Comment:
Not used anywhere yet outside tests -- this is just for illustration on how this will plug into authorization code.